org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/discovery/ZigBeeNodePropertyDiscoverer.java

```diff
@@ -185,7 +186,7 @@ public void setAlwaysUpdate(boolean alwaysUpdate) {
         if (otaCluster != null) {
             Integer fileVersion = otaCluster.getCurrentFileVersion(Long.MAX_VALUE);
             if (fileVersion != null) {
-                properties.put(Thing.PROPERTY_FIRMWARE_VERSION, String.format("%08X", fileVersion));
+                properties.put(Thing.PROPERTY_FIRMWARE_VERSION, String.format("%s%08X",HEX_PREFIX, fileVersion));
```

Comment:
I am aware that this code will be used as part of the device discovery and whenever the thing initialization takes place. However, the `Thing.PROPERTY_FIRMWARE_VERSION` property will be also modified after successfully updating the device firmware (see `ZigBeeThingHandler#updateFirmware`).

Reply:
So what you are saying is that there is still one point in the code missing that needs to be adapted in this PR, right?

Reply:
Yes and therefore, it might probably make sense to extract the constants as well.

Reply:
OK, thanks @hanneshofmann ! Regarding the constants, I still think that `0x` is something that does not need a constant - but I am fine with it :)